frontend/lib/src/components/elements/Spinner/Spinner.tsx

```diff
@@ -34,10 +35,49 @@ export interface SpinnerProps {
   element: SpinnerProto
 }
 
+// Formats the time in a human-readable way.
+// For example, 1.1 -> (1.1 seconds)
+// 65.3 -> (1 minute, 5.3 seconds)
+// 3661.1 -> (1 hour, 1 minute, 1.1 seconds)
+// TODO: In the future, we might want to replace this with `Intl.DurationFormat` (see
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat).
+// But that API is not available on Firefox yet.
```

Comment:
Instead of single line comments, its would be preferred to use a multi-line docstring comment here, e.g.:

https://github.com/streamlit/streamlit/blob/27fbf5eaee4c23d56f2b17bc10b5a8ee8ec941f6/frontend/lib/src/components/widgets/DataFrame/columns/utils.ts#L446-L456